src/chainparams.cpp

```diff
@@ -334,9 +335,15 @@ CChainParams* CChainParams::Factory(CBaseChainParams::Network network, CScript s
     }
 }
 
+CChainParams &Params(CBaseChainParams::Network network)
+{
+    globalSwitchingChainParams.reset(CChainParams::Factory(network, CScript()));
+    return *globalSwitchingChainParams;
+}
+
```

Comment:
Both can return a const ref though.
I mean, Params(network) is documented as deprecated here and bip70 is not implemented in the most ideal way with respect to chainparams. But those are things that should be fixed on bitcoin core and alpha-0.12/beta-0.12, not in alpha-0-10. Is anybody using the payment server with alpha anyway?
I really think just leaving Params(network) there until it's removed in bitcoin core is simpler for rebases. Maybe you or @TheBlueMatt disagree.